app/hooks/useWalletConnection.ts

```diff
@@ -0,0 +1,160 @@
+import { useState, useEffect } from "react";
+import { detectWallets } from "@/utils/wallet/detectedWallets";
+import { chainConfig } from "@/utils/wallet/chainConfig";
+import { useConnectors, useSwitchChain } from "wagmi";
+
+export function useWalletConnection(
+  setWalletState: any,
+  toggleWalletModal: () => void,
+) {
+  const { switchChain } = useSwitchChain();
+  const ethConnectors = useConnectors();
+
+  const [selectedChain, setSelectedChain] = useState<string | null>("bitcoin");
+  const [detectedWallets, setDetectedWallets] = useState<WalletOption[]>([]);
+
+  useEffect(() => {
+    const wallets = detectWallets(ethConnectors);
+    const walletsWithIcons = wallets
+      .map((detectedWallet) => {
+        for (const chain of chainConfig) {
+          const matchingWallet = chain.wallets.find(
+            (w) => w.id === detectedWallet.id,
+          );
+          if (matchingWallet) {
+            return {
+              ...detectedWallet,
+              icon: matchingWallet.icon,
+              name: matchingWallet.name,
+              downloadUrl: matchingWallet.downloadUrl,
+            };
+          }
+        }
+        return null;
+      })
+      .filter((w): w is NonNullable<typeof w> => w !== null);
+
+    setDetectedWallets(walletsWithIcons);
+  }, []);
+
+  const handleConnect = async (wallet: WalletOption) => {
+    if (!selectedChain) {
+      console.error("No chain selected.");
+      return;
+    }
+
+    try {
+      const detectedWalletForChain = detectedWallets.find((w) => {
```

Comment:
I recommend using a map for the `detectedWalletForChain`

Something like this should do the job:
```
const chainIdentifiers = {
  solana: "solana",
  kujira: "kujira", 
  cosmos: "cosmos",
  thorchain: "thorchain",
  mayachain: "maya",
  litecoin: "ltc",
  dogecoin: "doge",
  bitcoincash: "bch",
  bitcoin: "utxo"
};

const detectedWalletForChain = detectedWallets.find(w => {
  const identifier = chainIdentifiers[selectedChain];
  
  if (!identifier) {
    return w.id.split("-")[0] === wallet.id.split("-")[0];
  }
  
  return (w.id.includes(identifier) || wallet.id.includes(identifier)) && 
         w.id === wallet.id;
});
```

Reply:
Good job spotting this, I  wanted to change this implementation, thanks